### PR TITLE
API-24535 - fix status v1 response to match ruby

### PR DESCRIPTION
**Pull request by aaronhietpas**
Made separate V1 and V2 response bodies,  moved v1 to its own controller method and service method. Refactored the get emis and vaprofile request methods to return the response from each service instead of building the full Status response body.

### CODE REVIEW

File: veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/VeteranStatusIT.java [hunk 1]
```diff
@@ -34,18 +35,19 @@ static void assumeEnvironment() {
...
 
   @Test
   void veteranVerificationStatusFound() {
     String request = String.format("v1/status/%s", CONFIRMED_STATUS_ICN_DEFAULT);
     ExpectedResponse response = veteranVerificationGetRequest(request, "application/json", 200);
-    response.expectValid(VeteranStatusResponse.class);
-    VeteranStatusResponse status = response.response().getBody().as(VeteranStatusResponse.class);
+    response.expectValid(VeteranStatusResponseV2.class);
+    VeteranStatusResponseV2 status =
+        response.response().getBody().as(VeteranStatusResponseV2.class);
```
Comment (dlennon-va) [inline veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/VeteranStatusIT.java:50]: Should these be v1?

File: veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/VeteranStatusOauthIT.java [hunk 2]
```diff
@@ -46,8 +47,9 @@ void veteranStatusNotConfirmed() {
     String token = client.requestToken().accessToken();
     ExpectedResponse response =
         veteranVerificationTokenGetRequest("v1/status", "application/json", 200, token);
-    response.expectValid(VeteranStatusResponse.class);
-    VeteranStatusResponse status = response.response().getBody().as(VeteranStatusResponse.class);
+    response.expectValid(VeteranStatusResponseV2.class);
```
Comment (dlennon-va) [inline veteran-verification-tests/src/test/java/gov/va/api/lighthouse/veteranverification/tests/VeteranStatusOauthIT.java:50]: Should this be v1?

File: veteran-verification/src/main/java/gov/va/api/lighthouse/veteranverification/service/StatusVerificationService.java [hunk 3]
```diff
@@ -158,34 +156,71 @@ private VeteranStatusResponse getVaProfileVeteranResponse(
...
+          STATUS_REQUEST,
+          statusRequestId,
+          RequestState.REQUEST_COMPLETED,
+          response.getData().getAttributes().getVeteranStatus().equals("confirmed")
+              ? RequestResult.COMPLETED_CONFIRMED
+              : RequestResult.COMPLETED_NOT_CONFIRMED,
+          String.format(
+              "call to va profile status endpoint succeeded. veteran confirmation status: %s",
+              response.getData().getAttributes().getVeteranStatus()));
+      return response;
```
Comment (dlennon-va) [inline veteran-verification/src/main/java/gov/va/api/lighthouse/veteranverification/service/StatusVerificationService.java:219]: Could we simplify all this and avoid a bunch of duplication by having the Service class just return the V2 version of the object, and then the controller for just the V1 endpoint would take the V2 object and mold it into a V1
So  the controller would look something like this
```
@GET("/v1/status/{icn}")
public VeteranStatusResponseV1 getStatusV1(String icn) {
   return VeteranStatusResponseV1.from(statusVerificationService.veteranStatusVerificationGetResponseV2(icn));
}
```
And the VeteranStatusV1 model can have a `from` method that takes in a V2 object and builds a V1 object from the data inside it.

Comment (aaronhietpas) [reply veteran-verification/src/main/java/gov/va/api/lighthouse/veteranverification/service/StatusVerificationService.java:219]: I like that idea, making that change

**Review by dlennon-va — APPROVED** [review]

**Pull request merged**